#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: fix insecure temporary file creation in  catchsegv.sh
# DP: Related bugs: Debian #278278, Warty #2743, CAN-2004-0968
# DP: Upstream status: In CVS

PATCHLEVEL=1

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p$PATCHLEVEL < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p$PATCHLEVEL < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
--- glibc-2.3.2/debug/catchsegv.sh	2003-02-06 04:03:51.000000000 +0100
+++ glibc-2.3.2.new/debug/catchsegv.sh	2004-10-26 11:56:54.661026440 +0200
@@ -49,9 +49,7 @@
   esac
 fi
 
-segv_output=`basename "$prog"`.segv.$$
-# Make sure this output file does not exist.
-rm -f "$segv_output"
+segv_output=`mktemp \`basename "$prog".segv.XXXXXX\`` || exit 1
 
 # Redirect stderr to avoid termination message from shell.
 (exec 3>&2 2>/dev/null
